src/mame/konami/tmnt2.cpp

```diff
@@ -2499,6 +2501,7 @@ void glfgreat_state::glfgreat(machine_config &config)
 	m_k053260->add_route(0, "speaker", 1.0, 1);
 	m_k053260->add_route(1, "speaker", 1.0, 0);
 	m_k053260->sh1_cb().set(FUNC(glfgreat_state::z80_nmi_w));
+	m_k053260->tim2_cb().set(FUNC(glfgreat_state::z80_int_w));
```

Comment:
Not trampoline needed, suggested:
m_k053260->tim2_cb().set_inputline(m_audiocpu, 0, HOLD_LINE);

FYI what you have at z80_int_w will write it much more often than 2ms, and will never clear IRQ.
The correct way over there would be: if (state) m_audiocpu->set_input_line(0, HOLD_LINE);

Reply:
I have modified it as per your suggestion:
```C++
m_k053260->tim2_cb().set_inputline(m_audiocpu, 0, HOLD_LINE);
```
I have deleted the `z80_int_w` helper function. Should this produce the intended effect?

Reply:
Yes.

Reply:
Wait, isn't it supposed to clear the line with state == 0?

Reply:
jotego said nope. Only when Z80 takes the IRQ.

Reply:
There is a flip flop that latches the interrupt request until the Z80 serves it. See device **2E**, the CLR pin is controlled by the Z80's IORQ pin only.